pool/web/view/component/sortable_table.ml

```diff
@@ -0,0 +1,94 @@
+open Tyxml.Html
+
+let hx_get ~url ~target =
+  [ a_user_data "hx-get" url
+  ; a_user_data "hx-push-url" url
+  ; a_user_data "hx-target" target
+  ; a_user_data "hx-swap" "outerHTML"
+  ]
+;;
+
+type sort =
+  { url : Uri.t
+  ; query : Query.t
+  ; language : Pool_common.Language.t
+  }
+
+type col =
+  [ `column of Query.Column.t
+  | `custom of [ | Html_types.flow5 ] Tyxml_html.elt
+  ]
+
+let direction query column =
+  let open Query in
+  let field = Column.field column in
+  let is_selected =
+    match query.sort with
+    | Some sort -> Column.field Sort.(sort.column) = field
+    | None -> false
+  in
+  let direction =
+    let open Query.Sort in
+    match query.sort with
+    | Some sort -> Sort.(sort.order)
+    | None -> SortOrder.Ascending
+  in
+  if is_selected then Query.Sort.SortOrder.flip direction else direction
+;;
+
+let make_url { url; query; _ } column =
+  let query =
+    query
+    |> Query.with_sort_order (direction query column)
+    |> Query.with_sort_column column
+    |> Query.to_uri_query
+  in
+  let url = Uri.with_query url query in
+  Format.asprintf "%a" Uri.pp url
+;;
+
+let make_name { language; _ } col =
+  span [ Component_table.field_to_txt language (Query.Column.field col) ]
+;;
+
+let sort_icon { query; _ } col =
```

Comment:
@leostera Currently the table always shows icons for each sortable column. IMO it should only show an icon when the sort is applied to that column.

To define which direction to use to build the URL, I suggest using a default direction (ASC) for columns that currently do not have any sort applied.

Let me know if you have questions.

Reply:
@timohuber shoul be fixed in https://github.com/uzh/pool/pull/275/commits/1513c5b515dc6666b4bbb58faf55cb7b7b7eb207 🙌🏼

Reply:
Thanks!